### PR TITLE
129 edit projected budgets

### PR DESCRIPTION
**Pull request by mathiashoeld**
* Enable edit of projected budgets

* Enable edit of subproject currency

### CODE REVIEW

File: api/src/service/domain/workflow/subproject_updated.ts [hunk 1]
```diff
@@ -111,6 +112,8 @@ export function apply(
     ...(update.displayName !== undefined && { displayName: update.displayName }),
     // Only updated if defined in the `update`:
     ...(update.description !== undefined && { description: update.description }),
+    // Only updated if defined in the `update`:
+    ...(update.currency !== undefined && { currency: update.currency }),
```
[the remark this answers is not in the document]
Comment (mathiashoeld) [reply api/src/service/domain/workflow/subproject_updated.ts]: reverted commit 3543373f8659bddeef16752c117268661ce0e74b

[the remark this answers is not in the document]
Comment (mathiashoeld) [reply api/src/service/domain/workflow/subproject_updated.ts]: removed currency from http request and in the workflow layer, see 00d46442fdc8c3b70af189f461ea596d00e7f816

File: api/src/service/domain/workflow/project_projected_budget_update.ts [hunk 2]
```diff
@@ -56,6 +57,7 @@ export async function updateProjectedBudget(
   if (Result.isErr(result)) {
     return new InvalidCommand(ctx, budgetUpdated, [result]);
   }
+  logger.fatal(result);
```
Comment (kevinbader) [inline api/src/service/domain/workflow/project_projected_budget_update.ts]: debug?

File: api/src/service/domain/workflow/project_projected_budget_updated.ts [hunk 3]
```diff
@@ -86,13 +87,15 @@ export function mutate(project: Project.Project, event: Event): Result.Type<void
...
     x => x.organization === event.organization && x.currencyCode === event.currencyCode,
   );
 
-  if (targetBudget !== undefined) {
+  const budgetAlreadyExists = budgetToUpdateIndex !== -1;
+
+  if (budgetAlreadyExists) {
     // Update an existing budget:
-    targetBudget.value = event.value;
+    projectedBudgets[budgetToUpdateIndex].value = event.value;
```
Comment (kevinbader) [inline api/src/service/domain/workflow/project_projected_budget_updated.ts]: sry I don't see why this is better ^^

File: api/src/service/domain/workflow/subproject_projected_budget_updated.ts [hunk 4]
```diff
@@ -91,13 +93,14 @@ export function mutate(subproject: Subproject.Subproject, event: Event): Result.
...
+  const budgetToUpdateIndex = projectedBudgets.findIndex(
     x => x.organization === event.organization && x.currencyCode === event.currencyCode,
   );
+  const budgetAlreadyExists = budgetToUpdateIndex !== -1;
 
-  if (targetBudget !== undefined) {
+  if (budgetAlreadyExists) {
     // Update an existing budget:
-    targetBudget.value = event.value;
+    projectedBudgets[budgetToUpdateIndex].value = event.value;
```
Comment (kevinbader) [inline api/src/service/domain/workflow/subproject_projected_budget_updated.ts]: same here, I feel the previous version is actually more readable

File: api/src/service/domain/workflow/project_projected_budget_update.ts (hunk 2 above)
Comment (mathiashoeld) [reply api/src/service/domain/workflow/project_projected_budget_update.ts]: yep. removed

File: api/src/service/domain/workflow/project_projected_budget_updated.ts (hunk 3 above)
Comment (mathiashoeld) [reply api/src/service/domain/workflow/project_projected_budget_updated.ts]: changed back

File: api/src/service/domain/workflow/subproject_projected_budget_updated.ts (hunk 4 above)
Comment (mathiashoeld) [reply api/src/service/domain/workflow/subproject_projected_budget_updated.ts]: changed back

File: frontend/src/pages/Common/Budget2.js [hunk 5]
```diff
@@ -32,18 +37,47 @@ export default class Budget extends React.Component {
 
   deleteBudgetFromList(projectedBudgets, deletedProjectedBudgets, budgetToDelete) {
     this.props.storeDeletedProjectedBudget(this.addBudget(deletedProjectedBudgets, budgetToDelete));
+    console.log(this.props.projectedBudgets);
```
Comment (Stezido) [inline frontend/src/pages/Common/Budget2.js]:
```suggestion
```

**Comment by mathiashoeld** [comment]
Closes #129

**Pull request merged**